### PR TITLE
BaSP-MR-09-30: Updated member

### PR DESCRIPTION
**Pull request by danielninfi**
Member - Update

### CODE REVIEW

File: src/controllers/member.js [hunk 1]
```diff
@@ -1,155 +1,39 @@
...
+      city,
+      postalCode,
+      isActive,
+      membership,
+    },
+    { new: true },
+  )
+    .then((result) => {
+      if (!result) {
+        return res.status(404).json({
```
Comment (NicoCagnina) [inline src/controllers/member.js:27]: this should be a 404

File: src/controllers/member.js [hunk 2]
```diff
@@ -1,155 +1,39 @@
...
-    res.status(201).json({ msg: 'Member updated', member });
-  } else {
-    res.status(404).json({ msg: `Member ID ${idParam} not found` });
-  }
-});
-
-export default apiMembers;
+      return res.status(201).json(result);
+    })
+    .catch((error) => res.status(400).json(error));
```
Comment (NicoCagnina) [inline src/controllers/member.js:34]: this should be 500

File: src/models/Member.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+const mongoose = require('mongoose');
+
+const { Schema } = mongoose;
+
+const memberSchema = new Schema({
+  id: {
```
Comment (NicoCagnina) [inline src/models/Member.js:6]: id shouldn't be part of the schema. mongo handles this automatically

File: src/models/Member.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  dni: {
+    type: Number,
+    required: true,
+  },
+  birthday: {
+    type: Date,
+    required: true,
+  },
+  phone: {
+    type: Number,
```
Comment (NicoCagnina) [inline src/models/Member.js:28]: this should be string

File: src/validations/member.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+const Joi = require('joi');
+
+const validateUpdate = (req, res, next) => {
+  const memberValidation = Joi.object({
+    firstName: Joi.string().min(3).max(50),
+    lastName: Joi.string().min(3).max(50),
+    dni: Joi.number().min(6).max(8),
```
Comment (NicoCagnina) [inline src/validations/member.js:7]: careful here. min and max in numbers refer to minimum value and maximum value, it does not refer to the length.

here you can do this:
```
dni: Joi.number()
  .positive()
  .integer().
  .min(10000000)
  .max(99999999)
```

File: src/validations/member.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    birthday: Joi.date().max('now').custom((value, helpers) => {
+      const age = new Date().getFullYear() - new Date(value).getFullYear();
+      if (age < 12) {
+        return helpers.error('date.minAge', { minAge: 12 });
+      }
+      return value;
+    }),
+    phone: Joi.number().min(8).max(12),
```
Comment (NicoCagnina) [inline src/validations/member.js:15]: same as dni

**Review by NicoCagnina — CHANGES_REQUESTED** [review]

File: src/controllers/member.js (hunk 1 above)
Comment (danielninfi) [reply src/controllers/member.js:27]: is a 404, or should it be a 400?

File: src/validations/member.js (hunk 6 above)
Comment (danielninfi) [reply src/validations/member.js:15]: I am not using any regex, should I?

File: src/validations/member.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        return helpers.error('date.minAge', { minAge: 12 });
+      }
+      return value;
+    }),
+    phone: Joi.number()
+      .positive()
+      .integer()
+      .min(10000000)
+      .max(99999999),
+    email: Joi.string().email(),
```
Comment (JuanMornik) [inline src/validations/member.js:24]: add a regex here

File: src/controllers/member.js (hunk 1 above)
Comment (JuanMornik) [reply src/controllers/member.js:27]: it's a 404 error because the member doesn't exists

**Review by JuanMornik — CHANGES_REQUESTED** [review]

File: src/validations/member.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
+const Joi = require('joi');
+
+const validateUpdate = (req, res, next) => {
+  const memberValidation = Joi.object({
+    firstName: Joi.string().min(3).max(50),
```
Comment (JuanMornik) [inline src/validations/member.js:5]: add the .required() validation according to Members schema

File: src/validations/member.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+      .min(10000000)
+      .max(99999999),
+    birthday: Joi.date().max('now').custom((value, helpers) => {
+      const age = new Date().getFullYear() - new Date(value).getFullYear();
+      if (age < 12) {
+        return helpers.error('date.minAge', { minAge: 12 });
+      }
+      return value;
+    }),
+    phone: Joi.number()
```
[the remark this answers is not in the document]
Comment (danielninfi) [reply src/validations/member.js:19]: apologies, I added what was suggested but this slipped my mind.

File: src/validations/member.js (hunk 7 above)
Comment (danielninfi) [reply src/validations/member.js:24]: clarified in slack

File: src/models/Member.js [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  firstName: {
+    type: String,
+    required: true,
+  },
+  lastName: {
+    type: String,
+    required: true,
+  },
+  dni: {
+    type: Number,
```
Comment (NicoCagnina) [inline src/models/Member.js:15]: this should have min and max length

File: src/models/Member.js [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+    required: true,
+  },
+  birthday: {
+    type: Date,
+    required: true,
+  },
+  phone: {
+    type: String,
```
Comment (NicoCagnina) [inline src/models/Member.js:23]: this should have min and max length

**Review by NicoCagnina — CHANGES_REQUESTED** [review]

File: src/validations/member.js (hunk 5 above)
Comment (NicoCagnina) [reply src/validations/member.js]: yes!

File: src/validations/member.js (hunk 6 above)
Comment (NicoCagnina) [reply src/validations/member.js]: I don't think it's necessary.

**Review by NicoCagnina — APPROVED** [review]

**Review by JuanMornik — APPROVED** [review]

**Pull request merged**